backend/api/swagger.yaml

```diff
@@ -0,0 +1,1917 @@
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "401":
+          description: Authentication failed
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "403":
+          description: Insufficient permissions
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "404":
+          description: Resource not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+      security:
+      - bearerAuth: []
+    post:
+      tags:
+      - Kubernetes Resources
+      summary: Create a new resource
+      description: Creates a new resource of the specified type in the given namespace.
+      operationId: createNamespacedResource
+      parameters:
+      - name: resourceType
+        in: path
+        description: Type of the Kubernetes resource.
+        required: true
+        style: simple
+        explode: false
+        schema:
+          type: string
+          enum:
+          - Pod
+          - Deployment
+          - ConfigMap
+          - Secret
+          - Ingress
+          - PersistentVolumeClaim
+          - ReplicaSet
+          - StatefulSet
+          - DaemonSet
+          - Job
+          - CronJob
+          - Service
+          - ServiceAccount
+      - name: namespace
+        in: path
+        description: Name of the namespace.
+        required: true
+        style: simple
+        explode: false
+        schema:
+          type: string
+      requestBody:
+        description: JSON object representing the resource to be created.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Resource'
+        required: true
+      responses:
+        "201":
+          description: Resource created successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Resource'
+        "400":
+          description: Invalid input
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "401":
+          description: Authentication failed
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "403":
+          description: Insufficient permissions
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "404":
+          description: Resource not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+      security:
+      - bearerAuth: []
+  /k8s/namespaced/{resourceType}/{namespace}/{resourceName}:
```

Comment:
I think we don't need to include resourceType in this request path. According to kubernetes resourceNames must be unique within namespace even if they have different type. Also this helps if we will have any type of search functionality for a resource without knowing its type.

Reply:
![image](https://github.com/user-attachments/assets/efd04916-0de1-49b9-be71-903ab470647b)